protocols/virtualfund/virtualfund.go

```diff
@@ -75,6 +75,28 @@ func (c *Connection) insertGuaranteeInfo(a0 types.Funds, b0 types.Funds, vId typ
 	return nil
 }
 
+// handleProposal recieves a signed proposal and acts according to the leader / follower
+// status of the Connection's ConsensusChannel
+func (c *Connection) handleProposal(sp consensus_channel.SignedProposal) error {
+	if c == nil {
+		return fmt.Errorf("nil connection should not handle proposals")
+	}
+
+	if sp.Proposal.ChannelID != c.ConsensusChannel.Id {
+		return consensus_channel.ErrIncorrectChannelID
+	}
+
+	if c.ConsensusChannel.IsFollower() {
```

Comment:
Seems like `consensus_channel` has enough information internally to determine how to react to a signed proposal. Would it be better to encapsulate this `isFollower`/`isLeader` logic inside the `consensus_channel` struct logic (instead of exposing this logic to the virtual fund protocol)?